src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/session/DragAndDropSessionAction.java

```diff
@@ -0,0 +1,160 @@
+        this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public void getAndValidateEntities() {
+        session  = sessionRepository.findOne(dragAndDropSessionRequest.getSessionId());
+        hearingParts = session.getHearingParts();
+        validateIfJudgeCanBeChanged(session, dragAndDropSessionRequest);
+    }
+
+    @Override
+    public void act() {
+        try {
+            currentSessionAsString = objectMapper.writeValueAsString(session);
+        } catch (JsonProcessingException e) {
+            throw new SnlEventsException("Given session couldn't be converted into string");
+        }
+
+        Session session = updateSession(dragAndDropSessionRequest);
+        entityManager.detach(session);
+        session.setVersion(dragAndDropSessionRequest.getVersion());
+
+        sessionRepository.save(session);
+    }
+
+    @Override
+    public List<FactMessage> generateFactMessages() {
+        SessionWithHearingPartsFacts sessionWithHpFacts =
+            factsMapper.mapDbSessionToRuleJsonMessage(session, hearingParts);
+        List<FactMessage> facts = sessionWithHpFacts.getHearingPartsFacts().stream()
+            .map(hpFact -> new FactMessage(RulesService.UPSERT_HEARING_PART, hpFact))
+            .collect(Collectors.toList());
+        facts.add(new FactMessage(RulesService.UPSERT_SESSION, sessionWithHpFacts.getSessionFact()));
+
+        return facts;
+    }
+
+    @Override
+    public List<UserTransactionData> generateUserTransactionData() {
+        List<UserTransactionData> userTransactionDataList = new ArrayList<>();
+        userTransactionDataList.add(new UserTransactionData("session",
+            session.getId(),
+            currentSessionAsString,
+            "update",
+            "update",
+            0)
+        );
+
+        for (HearingPart hp : hearingParts) {
+            String hearingPartAsString = null;
+
+            try {
+                hearingPartAsString = objectMapper.writeValueAsString(hp);
+            } catch (JsonProcessingException e) {
+                throw new SnlRuntimeException("Could not convert hearing part to String. Hearing part id: "
+                    + hp.getId());
+            }
+
+            UserTransactionData transactionData = new UserTransactionData("hearingPart",
+                hp.getId(),
+                hearingPartAsString,
+                "lock",
+                "unlock",
+                0
+            );
+            userTransactionDataList.add(transactionData);
+        }
+
+        return userTransactionDataList;
+    }
+
+    @Override
+    public UUID getUserTransactionId() {
+        return dragAndDropSessionRequest.getUserTransactionId();
+    }
+
+    @Override
+    public UUID[] getAssociatedEntitiesIds() {
+        List<UUID> associatedEntitiesIds = new ArrayList<>();
+        associatedEntitiesIds.add(dragAndDropSessionRequest.getSessionId());
+        hearingParts.forEach(hp -> associatedEntitiesIds.add(hp.getId()));
```

Comment:
Should we add hearings also?

Reply:
in this action we are changing only session properties. 
There is a stroy to sync session start time with hearing part start time so within this task we could think about it